framework/decode/dx12_object_info.h

```diff
@@ -197,11 +214,12 @@ struct DxObjectExtraInfo
 struct DxObjectInfo
 {
     // Standard info stored for all DX objects.
-    IUnknown*                          object{ nullptr };
-    format::HandleId                   capture_id{ format::kNullHandleId };
-    uint64_t                           ref_count{ 1 };
-    uint64_t                           extra_ref{ 0 };
-    std::unique_ptr<DxObjectExtraInfo> extra_info;
+    IUnknown*                                              object{ nullptr };
+    format::HandleId                                       capture_id{ format::kNullHandleId };
+    uint64_t                                               ref_count{ 1 };
+    uint64_t                                               extra_ref{ 0 };
+    std::unique_ptr<DxObjectExtraInfo>                     extra_info;
+    std::unordered_map<VariableLengthArrayIndices, size_t> array_counts;
```

Comment:
Will there ever be more than one entry in this map?

Reply:
Yes, currently ID3D12Device5 could have up to 2 entries and ID3D12InfoQueue can have up to 3 entries. D3D11 support would also include up to 8 entries for ID3D11DeviceContext, 2 for ID3D11Device, and 2 for ID3D11ClassInstance.